pkg/k8shandler/curation.go

```diff
@@ -164,18 +164,23 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateCuratorConfigMap() er
 }
 
 func (clusterRequest *ClusterLoggingRequest) createOrUpdateCuratorSecret() error {
-
-	curatorSecret := NewSecret(
-		"curator",
-		clusterRequest.Cluster.Namespace,
-		map[string][]byte{
+	var secrets = map[string][]byte{}
+	Syncronize(func() error {
+		secrets = map[string][]byte{
```

Comment:
We're not checking for possible errors from GetWorkingDirFileContents()

Reply:
I think this is OK not to check missing here.  We need a bunch of secrets and we should fail and reconcile again.  The cert gen code should regen if something is missing.  If it does not that is a bug in the regen code IMO

Reply:
But if we don't check we don't fail and stick empty buffers into secrets.

Reply:
What is the action we can take in this situation?  What do we believe will change during a subsequent reconciliation that would resolve the issue?  The best we can do is really generate a log message and/or update status.  This particular file is probably moot as we want to remove curation from CLO but  if we are to do anything then it should be applied to the other places where we take similar action

Reply:
We can abort, refuse to update the secret and potentially make our situation worse. GetWorkingDirFileContents() is not very nuanced in how it reports errors, but there can be a genuine I/O failure that will make GetWorkingDirFileContents() return nil while there's a good cert in the secret as well as on disk. By proceeding we make things worse by wiping out a good cert from a secret. We will recover in time, but this is a disruption we can avoid.